src/org/thoughtcrime/securesms/loki/protocol/ClosedGroupsProtocolV2.kt

```diff
@@ -87,7 +90,154 @@ object ClosedGroupsProtocolV2 {
     }
 
     @JvmStatic
-    public fun leave(context: Context, groupPublicKey: String): Promise<Unit, Exception> {
+    fun explicitLeave(context: Context, groupPublicKey: String): Promise<Unit, Exception> {
+        val deferred = deferred<Unit, Exception>()
+        ThreadUtils.queue {
+            val userPublicKey = TextSecurePreferences.getLocalNumber(context)
+            val apiDB = DatabaseFactory.getLokiAPIDatabase(context)
+            val groupDB = DatabaseFactory.getGroupDatabase(context)
+            val groupID = doubleEncodeGroupID(groupPublicKey)
+            val group = groupDB.getGroup(groupID).orNull()
+            val updatedMembers = group.members.map { it.serialize() }.toSet() - userPublicKey
+            val admins = group.admins.map { it.serialize() }
+            val name = group.title
+            if (group == null) {
+                Log.d("Loki", "Can't leave nonexistent closed group.")
+                return@queue deferred.reject(Error.NoThread)
+            }
+            // Send the update to the group
+            @Suppress("NAME_SHADOWING")
+            val job = ClosedGroupUpdateMessageSendJobV2(groupPublicKey, ClosedGroupUpdateMessageSendJobV2.Kind.Leave)
+            job.setContext(context)
+            job.onRun() // Run the job immediately
+            // Remove the group private key and unsubscribe from PNs
+            disableLocalGroupAndUnsubscribe(context, apiDB, groupPublicKey, groupDB, groupID, userPublicKey)
+            // Mark the group as inactive
+            groupDB.setActive(groupID, false)
+            groupDB.removeMember(groupID, Address.fromSerialized(userPublicKey))
+            // Notify the PN server
+            LokiPushNotificationManager.performOperation(context, ClosedGroupOperation.Unsubscribe, groupPublicKey, userPublicKey)
```

Comment:
This shouldn't be necessary because you're already calling `disableLocalGroupAndUnsubscribe`